### ISSUE TITLE
DNS entries are not being automatically created on clients join

### ISSUE THREAD

**Issue by celso-alexandre** [Fmstrat/samba-domain#16]
Hi,

First, I must say that this docker image is very robust, really, thanks for sharing it.
I'm having a problem in which dns entries are not being automatically created when windows clients join the domain.

I've run my container to serve as PDC for a new domain.

I've build the image and created the container with the following commands:

`mkdir -p /data/docker/builds
cd /data/docker/builds
git clone https://github.com/Fmstrat/samba-domain.git
cd samba-domain
docker build -t samba-domain .

docker run -t -i \
        -e "DOMAIN=CORP.LOCAL" \
        -e "DOMAINPASS=P@ssw0rd" \
        -e "DNSFORWARDER=8.8.8.8" \
        -e "HOSTIP=192.168.1.3" \
        -p 192.168.1.3:53:53 \
        -p 192.168.1.3:53:53/udp \
        -p 192.168.1.3:88:88 \
        -p 192.168.1.3:88:88/udp \
        -p 192.168.1.3:135:135 \
        -p 192.168.1.3:137-138:137-138/udp \
        -p 192.168.1.3:139:139 \
        -p 192.168.1.3:389:389 \
        -p 192.168.1.3:389:389/udp \
        -p 192.168.1.3:445:445 \
        -p 192.168.1.3:464:464 \
        -p 192.168.1.3:464:464/udp \
        -p 192.168.1.3:636:636 \
        -p 192.168.1.3:1024-1044:1024-1044 \
        -p 192.168.1.3:3268-3269:3268-3269 \
        -v /etc/localtime:/etc/localtime:ro \
        -v /docker/samba/var/lib/samba:/var/lib/samba \
        -v /docker/samba/etc/samba/external:/etc/samba/external \
        --dns-search corp.local \
        --dns 192.168.1.3 \
        --add-host samba-dc.corp.local:192.168.1.3 \
        -h samba-dc \
        --name samba-dc \
        --privileged \
        samba-domain`

Is it expected to happen?

**Comment by Fmstrat** [comment]
Thanks for the feedback!

You know, I'm not sure. The most I know about client DNS management is what is in the Wiki: https://wiki.samba.org/index.php/DNS_Administration and I've never used client DNS aspects. My guess is Samba4 supports it, and there is probably something that needs to be included in the config for it to take place. I'm just unaware of what that is.

**Comment by Fmstrat** [comment]
Perhaps you could look into the reverse zone https://wiki.samba.org/index.php/Setting_up_Samba_as_an_Active_Directory_Domain_Controller#Create_a_reverse_zone ? Maybe that's it. The Samba team in IRC or their email list can probably help. If you do figure it out, I'd be happy to patch it in.

**Comment by celso-alexandre** [comment]
I have an old compiled from source Samba PDC that I've configured 5 years ago where it works. I will research this, and as soon I discover it, I will share it here, but maybe not so soon, because of a business project.

**Comment by celso-alexandre** [comment]
Man, sorry I've returned that late with a solution.
I discovered that the ubuntu repositories samba version were just too old (4 or more years old). So, I've made 2 forks of your repository, but not for the same use as yours, not exactly at least.
One is the same as yours, but with webmin, to help admnistering without too much workaround to get it going. The other one is focused on migrating a compiled samba version (is my case) to a docker container, and it worked like a charm, really. 
For anyone experiencing the same problem, just use a new samba compiled version, but it may require another docker image with a provision approach. Maybe changing the ubuntu version on Dockerfile to :latest could fix it too.
